osbuild/formats_common.py

```diff
@@ -0,0 +1,45 @@
+"""
+Common procedures and classes between the v1 and v2 formats
+"""
+
+from abc import abstractmethod
+
+
+class OSbuildError(Exception):
+    """
+    All exceptions that want to be formated in json for a user program must overload OSbuildError.
+    They also must be declared in the map 'map_error_id'.
+    """
+
+    @abstractmethod
+    def payload_as_dict(self):
+        pass
+
+    @property
+    def ID(self):
+        return OSbuildError.map_error_id.get(self.__class__, 0)
+
+    map_error_id = {}
+
+    @staticmethod
+    def add_error_id(error_klass):
+        OSbuildError.map_error_id[error_klass.__class__] = len(OSbuildError.map_error_id)+1
+
+
+# pylint: disable=inconsistent-return-statements #Given an exception, there always be something to catch and return
+# pylint: disable=broad-except #We need to catch all exceptions here to have a default return value
+def format_exception(error: Exception):
+    """produces a json output from an OSbuildError"""
+    try:
+        raise error
+    except OSbuildError as e:
+        return {
+            "id": e.ID,
+            "human-readable": f"{e}",
+            "payload": e.payload_as_dict()
+        }
+    except Exception as e:
+        return {
+            "id": 0,
+            "human-readable": f"{e}",
+        }
```

Comment:
I'm not sure I understand the benefit of using exceptions everywhere, especially given this function.  This function is called from the `except` block in `osbuild_cli` and then reraises and catches to format the response.  Why not just format the response immediately?  Is the plan to have this be a centralised exception formatting function that we call from everywhere?

Reply:
So I think @gicmo wanted to be able to catch any exception and format it there. So we discussed this pattern on slack, maybe he can explain more. For now it's only used to get a SourceException, so it makes it look overkill, but we are planning to add a bunch of oher ones here. The fallback to general Exception is to let OSBuild able to format an unknown one to json, when composer would load this unknown one, it would treat it as a Internal error. This kind of error meaning that we did not cover all the corner cases.

Reply:
This bit changed completely, I resolve the conversation.